hazelcast/src/main/java/com/hazelcast/nio/serialization/compact/CompactReader.java

```diff
@@ -376,6 +376,19 @@
     @Nullable
     OffsetDateTime readTimestampWithTimezone(@Nonnull String fieldName, @Nullable OffsetDateTime defaultValue);
 
+    /**
+     * Reads an enum.
+     *
+     * @param fieldName name of the field.
+     * @param enumType underlying enum type.
+     * @return the value of the field.
+     * @throws HazelcastSerializationException if the field does not exist in the schema
+     *                                         or the type of the field does not match
+     *                                         with the one defined in the schema.
+     */
+    @Nullable
+    <T extends Enum> T readEnum(@Nonnull String fieldName, Class<T> enumType);
```

Comment:
We decided to not give an API for enums/enum arrays in both the GenericRecord and CompactReader/Writer interfaces. For the CompactReader/Writer, we want to maintain a FieldType for each kind of object we support.

For the GenericRecord, there are more reasons:
- We want GenericRecord API to be free of any type of class as on the server-side, there might not be such class. After all, GenericRecord is designed to be used as a way to represent domain objects without any sort of classes.
- We also want to maintain FieldType/object mapping there.
- Also, we want GenericRecord to be a polyglot. You may represent enums with Strings in Java, but they might be integers in C++.We want to maintain consistency across different languages.

Therefore, could you please remove readEnum and readEnumArray from the CompactReader and writeEnum and writeEnumArray from the CompactWriter?

We just want to implicitly support enums/enum arrays for the reflective serializer. For the explicit serializer use cases (where you use CompactReader/Writer), we want them to represent enums in the way they want.

Reply:
Ach, I get it, "inline" this in serializer. Ok, will be done soon, sorry for misunderstanding

Reply:
Done